core/src/main/scala-3/chisel3/Module.scala

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.experimental.{BaseModule, SourceInfo}
+
+object Module extends ObjectModuleImpl with SourceInfoDoc {
+
+  /** A wrapper method that all Module instantiations must be wrapped in
+    * (necessary to help Chisel track internal state).
+    *
+    * @param bc the Module being created
+    *
+    * @return the input module `m` with Chisel metadata properly set
+    */
+  // TODO(adkian-sifive) the callsite here explicitly passes
+  // sourceInfo so it cannot be a contextual parameter
+  def apply[T <: BaseModule](bc: => T): T = _applyImpl(bc)
+  def do_apply[T <: BaseModule](bc: => T)(implicit sourceInfo: SourceInfo): T = apply(bc)
```

Comment:
```suggestion
  def apply[T <: BaseModule](bc: => T)(using SourceInfo): T = _applyImpl(bc)
```
No do_apply and apply needs the context parameter.